canonical-web-and-design/maas-ui
# fix(ui): standardise shape of node actions

## Done

- Standardised shape of node action creators (e.g. commission, deploy etc) so that the name of the keys always match what the API expects (`systemId` => `system_id`). I've kept `extra` only in the reducers since it was kind of annoying having to include it in all the action creators.
- Fixed a bug where commissioning didn't work from the in-table action menu

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the machine list and check that you can commission a machine from the table (from the Status column). It should dispatch an action without any `extra` details - i.e, with defaults.
- Check a few other in-table actions as well as actions from the action menu to see that they still work

## Fixes

Fixes #3658
